Distiller-server/src/main/java/com/mapr/distiller/server/producers/raw/ProcRecordProducer.java

```diff
@@ -0,0 +1,428 @@
+package com.mapr.distiller.server.producers.raw;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.TreeSet;
+import java.io.File;
+import java.net.NetworkInterface;
+import java.lang.Integer;
+import java.io.FilenameFilter;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import com.mapr.distiller.server.recordtypes.DiskstatRecord;
+import com.mapr.distiller.server.recordtypes.NetworkInterfaceRecord;
+import com.mapr.distiller.server.recordtypes.ProcessResourceRecord;
+import com.mapr.distiller.server.recordtypes.SystemCpuRecord;
+import com.mapr.distiller.server.recordtypes.SystemMemoryRecord;
+import com.mapr.distiller.server.recordtypes.TcpConnectionStatRecord;
+import com.mapr.distiller.server.recordtypes.ThreadResourceRecord;
+import com.mapr.distiller.server.scheduler.GatherMetricEvent;
+import com.mapr.distiller.server.scheduler.MetricEventComparator;
+import com.mapr.distiller.server.queues.RecordQueue;
+import com.mapr.distiller.server.queues.SubscriptionRecordQueue;
+
+public class ProcRecordProducer extends Thread {
+	//This should be set true when ProcRecordProducer thread should exit.
+	private boolean shouldExit=false;
+	
+	//This holds the list of metrics to gather sorted by the time at which they should be gathered.
+	private TreeSet<GatherMetricEvent> metricSchedule = new TreeSet<GatherMetricEvent>(new MetricEventComparator());
+	
+	//The number of clock ticks (jiffies) per second.  Typically 100 but custom kernels can be compiled otherwise
+	private int clockTick;
+	
+	//The length of time, in milliseconds, represented by a single clock tick (jiffy). Typically 10ms
+	private Double clockTickms;
```

Comment:
These "Double" can be "double" primitive types. Unless, we are using it as part of a List or Map we don't need it to be of wrapper type.